runtime/runc_jailer.go

```diff
@@ -270,10 +270,10 @@ func (j *runcJailer) BuildJailedRootHandler(cfg *config.Config, machineConfig *f
 	}
 }
 
-func (j *runcJailer) makeLinkInJail(src string) (string, error) {
+// makeLinkInJail creates a symlink to `src` inside the jail directory.
+func (j *runcJailer) makeLinkInJail(src, base string) (string, error) {
 	root := j.RootPath()
 
```

Comment:
Is it worth leaving the filepath.Base call but doing it on base instead of src to ensure we don't accidentally provide a path instead in the future here?

Reply:
Good point. How about returning an error when `filepath.Base(base) != base`? I agree that `base` shouldn't be a path here. But correcting that automatically may hide the issue rather than let developers know and fix that.

Reply:
Added `strings.ContainsRune(base, os.PathSeparator)` as a check.